test/malloc_compliance_tests_glibc.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// run basic compliance tests for glibc's functions (sanity check)
+
+#include "base.hpp"
+
+using umf_test::test;
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "../src/memory_pool_internal.h"
+#include "malloc_compliance_tests.hpp"
+#include <umf/memory_pool_ops.h>
+
+// casting to (uintptr_t) because of:
+// warning C4312: 'type cast': conversion from 'unsigned int' to 'void *' of greater size
+#define POOL_PRIV ((void *)(uintptr_t)0xDEADBEAF)
+
+static void *glibc_malloc(void *pool, size_t size) {
```

Comment:
FYI, we already have implementation of a pool based on malloc: https://github.com/oneapi-src/unified-memory-framework/blob/main/test/common/pool.hpp#L112

And here it is used to instantiate the umfPoolTest: https://github.com/oneapi-src/unified-memory-framework/blob/55f63546d558394be3b09493e472f95652e2c722/test/memoryPoolAPI.cpp#L119

Reply:
So, as I suggested in another comment, I think you should just implement the tests from malloc_compliance_tests.*pp as `umfPoolTests` using TEST_P and it would already automatically run them on a malloc-based pool and on a disjoint_pool. You won't need this file at all.

Reply:
Thanks @igchor a lot!
Done,